pkg/awsutils/awsutils.go

```diff
@@ -501,15 +501,7 @@ func (cache *EC2InstanceMetadataCache) RefreshSGIDs(mac string) error {
 	cache.securityGroups.Set(sgIDs)
 
 	if !cache.useCustomNetworking && (addedSGsCount != 0 || deletedSGsCount != 0) {
-		allENIs, err := cache.GetAttachedENIs()
```

Comment:
We need to fill the ENIs from datastore, else the eniIDs will always be `null`

Reply:
Maybe this - https://github.com/aws/amazon-vpc-cni-k8s/blob/master/pkg/ipamd/datastore/data_store.go#L1192 or any other get function from datastore. Basically we will just use the ENIs from datastore instead of IMDS to sync SGs.

Reply:
@jayanthvn  - I have addressed the review comment, and add the test information. PTAL again. Thanks.